great_expectations/experimental/datasources/interfaces.py

```diff
@@ -283,6 +287,27 @@ def sort_batches(self, batch_list: List[Batch]) -> None:
                 ) from e
 
 
+def _sort_batches_with_none_metadata_values(
+    key: str,
+) -> Callable[[Batch, Batch], int]:
+    def _compare_function(a: Batch, b: Batch) -> int:
+        if a.metadata[key] is not None and b.metadata[key] is not None:
+            if a.metadata[key] < b.metadata[key]:
+                return -1
+            elif a.metadata[key] > b.metadata[key]:
+                return 1
+            else:
+                return 0
+        elif a.metadata[key] is None and b.metadata[key] is None:
```

Comment:
I know next to nothing about how our `metadata` works but should we accessing these values with `.get()`?
```python
if a.metadata.get(key) is None and b.metadata.get(key) is None:
```

Reply:
@billdirks I am just curious -- it this is a splitter-focused contribution, how come work on sorters is also included?  Thank you.

Reply:
The splitters determine which fields can be sorted on. Some entries in our non-datetime columns are null which can cause integration tests to fail. This fixes that bug so I can include the non-datetime splitters in the integration tests.

Reply:
Actually, it may have been a postgres dataset with nulls in it that hit this crasher (our integration tests currently only use a sqlite dataset). I don't recall which dataset it was.

Reply:
@Kilo59 The keys are determined by the splitter (in sql, in pandas/spark it is the regex) so should always be present. If they aren't then that is a bug.